## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 481933`
Commit: `e29b1261d02e4ed95636d4f57ed6e107e90417d2`

---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>dump1090-fa</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 40 packages built:</summary>
  <ul>
    <li>abracadabra</li>
    <li>cubicsdr</li>
    <li>dumphfdl</li>
    <li>gnss-sdr</li>
    <li>gnuradio</li>
    <li>gnuradio.man</li>
    <li>gnuradioMinimal</li>
    <li>gnuradioMinimal.man</li>
    <li>gnuradioPackages.bladeRF</li>
    <li>gnuradioPackages.fosphor</li>
    <li>gnuradioPackages.gr-difi</li>
    <li>gnuradioPackages.lora_sdr</li>
    <li>gnuradioPackages.lora_sdr.dev</li>
    <li>gnuradioPackages.osmosdr</li>
    <li>gnuradioPackages.osmosdr.dev</li>
    <li>gqrx</li>
    <li>gqrx-gr-audio</li>
    <li>gqrx-portaudio</li>
    <li>inspectrum</li>
    <li>openwebrx</li>
    <li>openwebrx.dist</li>
    <li>pothos</li>
    <li>python313Packages.soapysdr-with-plugins</li>
    <li>python314Packages.soapysdr-with-plugins</li>
    <li>qradiolink</li>
    <li>rtl_433</li>
    <li>sdrangel</li>
    <li>sdrpp</li>
    <li>sigdigger</li>
    <li>soapysdr-with-plugins</li>
    <li>soapyuhd</li>
    <li>srsran</li>
    <li>srsran.dev</li>
    <li>suscan</li>
    <li>trunk-recorder</li>
    <li>uhd</li>
    <li>uhd.dev</li>
    <li>uhdMinimal</li>
    <li>uhdMinimal.dev</li>
    <li>welle-io</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>dump1090-fa</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 39 packages built:</summary>
  <ul>
    <li>abracadabra</li>
    <li>cubicsdr</li>
    <li>dumphfdl</li>
    <li>gnss-sdr</li>
    <li>gnuradio</li>
    <li>gnuradio.man</li>
    <li>gnuradioMinimal</li>
    <li>gnuradioMinimal.man</li>
    <li>gnuradioPackages.bladeRF</li>
    <li>gnuradioPackages.fosphor</li>
    <li>gnuradioPackages.gr-difi</li>
    <li>gnuradioPackages.lora_sdr</li>
    <li>gnuradioPackages.lora_sdr.dev</li>
    <li>gnuradioPackages.osmosdr</li>
    <li>gnuradioPackages.osmosdr.dev</li>
    <li>gqrx</li>
    <li>gqrx-gr-audio</li>
    <li>gqrx-portaudio</li>
    <li>inspectrum</li>
    <li>openwebrx</li>
    <li>openwebrx.dist</li>
    <li>pothos</li>
    <li>python313Packages.soapysdr-with-plugins</li>
    <li>python314Packages.soapysdr-with-plugins</li>
    <li>qradiolink</li>
    <li>rtl_433</li>
    <li>sdrangel</li>
    <li>sdrpp</li>
    <li>sigdigger</li>
    <li>soapysdr-with-plugins</li>
    <li>soapyuhd</li>
    <li>srsran</li>
    <li>srsran.dev</li>
    <li>suscan</li>
    <li>trunk-recorder</li>
    <li>uhd</li>
    <li>uhd.dev</li>
    <li>uhdMinimal</li>
    <li>uhdMinimal.dev</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>gnuradio</li>
    <li>gnuradio.man</li>
    <li>gnuradioPackages.lora_sdr</li>
    <li>gnuradioPackages.lora_sdr.dev</li>
    <li>gnuradioPackages.osmosdr</li>
    <li>gnuradioPackages.osmosdr.dev</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>gnuradioMinimal</li>
    <li>gnuradioMinimal.man</li>
    <li>gqrx (gqrx-portaudio)</li>
    <li>gqrx-gr-audio</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>gnuradioPackages.osmosdr</li>
    <li>gnuradioPackages.osmosdr.dev</li>
    <li>gqrx (gqrx-portaudio)</li>
    <li>gqrx-gr-audio</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>gnuradio</li>
    <li>gnuradio.man</li>
    <li>gnuradioMinimal</li>
    <li>gnuradioMinimal.man</li>
    <li>gnuradioPackages.lora_sdr</li>
    <li>gnuradioPackages.lora_sdr.dev</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>dump1090-fa</summary>
<pre>@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
build flags: SHELL=/nix/store/lw117lsr8d585xs63kx5k233impyrq7q-bash-5.3p3/bin/bash DUMP1090_VERSION=10.2 showconfig dump1090 view1090 faup1090
Building with:
  Version string:   10.2
  Architecture:     x86_64
  DSP mix:          x86
  RTLSDR support:   yes
  BladeRF support:  yes
  HackRF support:   yes
  LimeSDR support:  yes
  SoapySDR support: yes
gcc  -I. -D_POSIX_C_SOURCE=200112L -DMODES_DUMP1090_VERSION=\&quot;10.2\&quot; -DMODES_DUMP1090_VARIANT=\&quot;dump1090-fa\&quot; -D_DEFAULT_SOURCE -DENABLE_CPUFEATURES -Icpu_features/include -DENABLE_RTLSDR -DENABLE_BLADERF -DENABLE_HACKRF -DENABLE_LIMESDR -DENABLE_SOAPYSDR -DSTARCH_MIX_X86 -O3 -g -std=c11 -fno-common -Wall -Wmissing-declarations -Werror -Wformat-signedness -W -I/nix/store/mhfn813bcl6zmdhfixqc532nh29pwm5f-rtl-sdr-blog-1.3.5-dev/include/ -I/nix/store/1qh3nd2iz9b42zjwsjxs482kdk2x7ccd-libbladeRF-2025.10/include/ -I/nix/store/rggzr6nc5jida5x16009cj6v8nqmjhrg-hackrf-2024.02.1/include -I/nix/store/rggzr6nc5jida5x16009cj6v8nqmjhrg-hackrf-2024.02.1/include/libhackrf -I/nix/store/ssxz4zrby6lxrcd15043jqdm0waiz8pk-libusb-1.0.29-dev/include/libusb-1.0 -I/nix/store/26ijwag856mby2pnclzwf62v2sx0715k-limesuite-23.11.0/include -I/nix/store/knjasbab448vagfvcdglvjgipm86ns03-soapysdr-0.8.1-unstable-2025-10-05-03/include -c dump1090.c -o dump1090.o
gcc  -I. -D_POSIX_C_SOURCE=200112L -DMODES_DUMP1090_VERSION=\&quot;10.2\&quot; -DMODES_DUMP1090_VARIANT=\&quot;dump1090-fa\&quot; -D_DEFAULT_SOURCE -DENABLE_CPUFEATURES -Icpu_features/include -DENABLE_RTLSDR -DENABLE_BLADERF -DENABLE_HACKRF -DENABLE_LIMESDR -DENABLE_SOAPYSDR -DSTARCH_MIX_X86 -O3 -g -std=c11 -fno-common -Wall -Wmissing-declarations -Werror -Wformat-signedness -W -I/nix/store/mhfn813bcl6zmdhfixqc532nh29pwm5f-rtl-sdr-blog-1.3.5-dev/include/ -I/nix/store/1qh3nd2iz9b42zjwsjxs482kdk2x7ccd-libbladeRF-2025.10/include/ -I/nix/store/rggzr6nc5jida5x16009cj6v8nqmjhrg-hackrf-2024.02.1/include -I/nix/store/rggzr6nc5jida5x16009cj6v8nqmjhrg-hackrf-2024.02.1/include/libhackrf -I/nix/store/ssxz4zrby6lxrcd15043jqdm0waiz8pk-libusb-1.0.29-dev/include/libusb-1.0 -I/nix/store/26ijwag856mby2pnclzwf62v2sx0715k-limesuite-23.11.0/include -I/nix/store/knjasbab448vagfvcdglvjgipm86ns03-soapysdr-0.8.1-unstable-2025-10-05-03/include -c anet.c -o anet.o
gcc  -I. -D_POSIX_C_SOURCE=200112L -DMODES_DUMP1090_VERSION=\&quot;10.2\&quot; -DMODES_DUMP1090_VARIANT=\&quot;dump1090-fa\&quot; -D_DEFAULT_SOURCE -DENABLE_CPUFEATURES -Icpu_features/include -DENABLE_RTLSDR -DENABLE_BLADERF -DENABLE_HACKRF -DENABLE_LIMESDR -DENABLE_SOAPYSDR -DSTARCH_MIX_X86 -O3 -g -std=c11 -fno-common -Wall -Wmissing-declarations -Werror -Wformat-signedness -W -I/nix/store/mhfn813bcl6zmdhfixqc532nh29pwm5f-rtl-sdr-blog-1.3.5-dev/include/ -I/nix/store/1qh3nd2iz9b42zjwsjxs482kdk2x7ccd-libbladeRF-2025.10/include/ -I/nix/store/rggzr6nc5jida5x16009cj6v8nqmjhrg-hackrf-2024.02.1/include -I/nix/store/rggzr6nc5jida5x16009cj6v8nqmjhrg-hackrf-2024.02.1/include/libhackrf -I/nix/store/ssxz4zrby6lxrcd15043jqdm0waiz8pk-libusb-1.0.29-dev/include/libusb-1.0 -I/nix/store/26ijwag856mby2pnclzwf62v2sx0715k-limesuite-23.11.0/include -I/nix/store/knjasbab448vagfvcdglvjgipm86ns03-soapysdr-0.8.1-unstable-2025-10-05-03/include -c interactive.c -o interactive.o
interactive.c: In function &#x27;interactiveShowData&#x27;:
interactive.c:143:23: error: initializer-string for array of &#x27;char&#x27; truncates NUL terminator but destination lacks &#x27;nonstring&#x27; attribute (5 chars into 4 available) [8;;https://gcc.gnu.org/onlinedocs/gcc-15.2.0/gcc/Warning-Options.html#index-Wno-unterminated-string-initialization-Werror=unterminated-string-initialization8;;]
  143 |     char spinner[4] = &quot;|/-\\&quot;;
      |                       ^~~~~~~
cc1: all warnings being treated as errors
make: *** [Makefile:223: interactive.o] Error 1</pre>
</details>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>dump1090-fa</summary>
<pre>Running phase: buildPhase
build flags: SHELL=/nix/store/q1lr86rzsxxbbqdz8m6bcv94saaz8wlv-bash-5.3p3/bin/bash DUMP1090_VERSION=10.2 showconfig dump1090 view1090 faup1090
Building with:
  Version string:   10.2
  Architecture:     aarch64
  DSP mix:          aarch64
  RTLSDR support:   yes
  BladeRF support:  yes
  HackRF support:   yes
  LimeSDR support:  yes
  SoapySDR support: yes
gcc  -I. -D_POSIX_C_SOURCE=200112L -DMODES_DUMP1090_VERSION=\&quot;10.2\&quot; -DMODES_DUMP1090_VARIANT=\&quot;dump1090-fa\&quot; -D_DEFAULT_SOURCE -DENABLE_CPUFEATURES -Icpu_features/include -DENABLE_RTLSDR -DENABLE_BLADERF -DENABLE_HACKRF -DENABLE_LIMESDR -DENABLE_SOAPYSDR -DSTARCH_MIX_AARCH64 -O3 -g -std=c11 -fno-common -Wall -Wmissing-declarations -Werror -Wformat-signedness -W -I/nix/store/nfj4x5wjmb31pr98hfrm1hp94r1q91yl-rtl-sdr-blog-1.3.5-dev/include/ -I/nix/store/r9h70crjz6g380fm68n18fihajvd535n-libbladeRF-2025.10/include/ -I/nix/store/c2xmf5mbqwa0mkr2w512592m673fn80z-hackrf-2024.02.1/include -I/nix/store/c2xmf5mbqwa0mkr2w512592m673fn80z-hackrf-2024.02.1/include/libhackrf -I/nix/store/md86n8l5isknzhy4kbi100zvyq99xl9i-libusb-1.0.29-dev/include/libusb-1.0 -I/nix/store/2kg98kv74l8d310pv31l27nmhj34lakg-limesuite-23.11.0/include -I/nix/store/sqp5grkpadbwzb2ay2778di0hxild6pz-soapysdr-0.8.1-unstable-2025-10-05-03/include -c dump1090.c -o dump1090.o
gcc  -I. -D_POSIX_C_SOURCE=200112L -DMODES_DUMP1090_VERSION=\&quot;10.2\&quot; -DMODES_DUMP1090_VARIANT=\&quot;dump1090-fa\&quot; -D_DEFAULT_SOURCE -DENABLE_CPUFEATURES -Icpu_features/include -DENABLE_RTLSDR -DENABLE_BLADERF -DENABLE_HACKRF -DENABLE_LIMESDR -DENABLE_SOAPYSDR -DSTARCH_MIX_AARCH64 -O3 -g -std=c11 -fno-common -Wall -Wmissing-declarations -Werror -Wformat-signedness -W -I/nix/store/nfj4x5wjmb31pr98hfrm1hp94r1q91yl-rtl-sdr-blog-1.3.5-dev/include/ -I/nix/store/r9h70crjz6g380fm68n18fihajvd535n-libbladeRF-2025.10/include/ -I/nix/store/c2xmf5mbqwa0mkr2w512592m673fn80z-hackrf-2024.02.1/include -I/nix/store/c2xmf5mbqwa0mkr2w512592m673fn80z-hackrf-2024.02.1/include/libhackrf -I/nix/store/md86n8l5isknzhy4kbi100zvyq99xl9i-libusb-1.0.29-dev/include/libusb-1.0 -I/nix/store/2kg98kv74l8d310pv31l27nmhj34lakg-limesuite-23.11.0/include -I/nix/store/sqp5grkpadbwzb2ay2778di0hxild6pz-soapysdr-0.8.1-unstable-2025-10-05-03/include -c anet.c -o anet.o
gcc  -I. -D_POSIX_C_SOURCE=200112L -DMODES_DUMP1090_VERSION=\&quot;10.2\&quot; -DMODES_DUMP1090_VARIANT=\&quot;dump1090-fa\&quot; -D_DEFAULT_SOURCE -DENABLE_CPUFEATURES -Icpu_features/include -DENABLE_RTLSDR -DENABLE_BLADERF -DENABLE_HACKRF -DENABLE_LIMESDR -DENABLE_SOAPYSDR -DSTARCH_MIX_AARCH64 -O3 -g -std=c11 -fno-common -Wall -Wmissing-declarations -Werror -Wformat-signedness -W -I/nix/store/nfj4x5wjmb31pr98hfrm1hp94r1q91yl-rtl-sdr-blog-1.3.5-dev/include/ -I/nix/store/r9h70crjz6g380fm68n18fihajvd535n-libbladeRF-2025.10/include/ -I/nix/store/c2xmf5mbqwa0mkr2w512592m673fn80z-hackrf-2024.02.1/include -I/nix/store/c2xmf5mbqwa0mkr2w512592m673fn80z-hackrf-2024.02.1/include/libhackrf -I/nix/store/md86n8l5isknzhy4kbi100zvyq99xl9i-libusb-1.0.29-dev/include/libusb-1.0 -I/nix/store/2kg98kv74l8d310pv31l27nmhj34lakg-limesuite-23.11.0/include -I/nix/store/sqp5grkpadbwzb2ay2778di0hxild6pz-soapysdr-0.8.1-unstable-2025-10-05-03/include -c interactive.c -o interactive.o
interactive.c: In function &#x27;interactiveShowData&#x27;:
interactive.c:143:23: error: initializer-string for array of &#x27;char&#x27; truncates NUL terminator but destination lacks &#x27;nonstring&#x27; attribute (5 chars into 4 available) [-Werror=unterminated-string-initialization]
  143 |     char spinner[4] = &quot;|/-\\&quot;;
      |                       ^~~~~~~
cc1: all warnings being treated as errors
make: *** [Makefile:223: interactive.o] Error 1</pre>
</details>
</details>

--- a/pkgs/applications/radio/gnuradio/wrapper.nix
+++ b/pkgs/applications/radio/gnuradio/wrapper.nix
@@ -50,6 +50,8 @@
     soapyplutosdr
     soapyremote
     soapyrtlsdr
+  ]
+  ++ lib.optionals (unwrapped.hasFeature "gr-uhd") [
     soapyuhd
   ],
   # Allow to add whatever you want to the wrapper

--- a/pkgs/by-name/th/thrift/package.nix
+++ b/pkgs/by-name/th/thrift/package.nix
@@ -85,7 +85,6 @@ stdenv.mkDerivation (finalAttrs: {
     # Tests that hang up in the Darwin sandbox
     "SecurityTest"
     "SecurityFromBufferTest"
-    "python_test"
 
     # fails on hydra, passes locally
     "concurrency_test"

--- a/pkgs/by-name/uh/uhd/package.nix
+++ b/pkgs/by-name/uh/uhd/package.nix
@@ -1,6 +1,7 @@
 {
   lib,
   stdenv,
+  llvmPackages_20,
   fetchurl,
   fetchFromGitHub,
   cmake,
@@ -34,9 +35,17 @@
 
 let
   inherit (lib) optionals cmakeBool;
+  stdenv' = (
+    # Fix a compilation issue on Darwin, that upstream is aware of:
+    # https://github.com/EttusResearch/uhd/issues/881
+    if stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isAarch64 then
+      llvmPackages_20.stdenv
+    else
+      stdenv
+  );
 in
 
-stdenv.mkDerivation (finalAttrs: {
+stdenv'.mkDerivation (finalAttrs: {
   pname = "uhd";
   # NOTE: Use the following command to update the package, and the uhdImageSrc attribute:
   #
@@ -147,7 +156,7 @@ stdenv.mkDerivation (finalAttrs: {
     # ABI differences GCC 7.1
     # /nix/store/wd6r25miqbk9ia53pp669gn4wrg9n9cj-gcc-7.3.0/include/c++/7.3.0/bits/vector.tcc:394:7: note: parameter passing for argument of type 'std::vector<uhd::range_t>::iterator {aka __gnu_cxx::__normal_iterator<uhd::range_t*, std::vector<uhd::range_t> >}' changed in GCC 7.1
   ]
-  ++ optionals stdenv.hostPlatform.isAarch32 [
+  ++ optionals stdenv'.hostPlatform.isAarch32 [
     "-DCMAKE_CXX_FLAGS=-Wno-psabi"
   ];
 
@@ -180,7 +189,7 @@ stdenv.mkDerivation (finalAttrs: {
   ];
 
   # many tests fails on darwin, according to ofborg
-  doCheck = !stdenv.hostPlatform.isDarwin;
+  doCheck = !stdenv'.hostPlatform.isDarwin;
 
   doInstallCheck = true;
 
@@ -191,7 +200,7 @@ stdenv.mkDerivation (finalAttrs: {
     "installFirmware"
     "removeInstalledTests"
   ]
-  ++ optionals (enableUtils && stdenv.hostPlatform.isLinux) [
+  ++ optionals (enableUtils && stdenv'.hostPlatform.isLinux) [
     "moveUdevRules"
   ];
 

